doc/source/advanced.md

```diff
@@ -39,7 +39,45 @@ We recommend the community-maintained [nginx-ingress](https://github.com/helm/ch
 controller, [**kubernetes/ingress-nginx**](https://github.com/kubernetes/ingress-nginx).
 Note that Nginx maintains two additional ingress controllers.
 For most use cases, we recommend the community maintained **kubernetes/ingress-nginx** since that
-is the ingress controller that the development team has the most experience using.
+is the ingress controller that the development team has the most experience using. See below for
+**GCE Ingress** settings.
+
+#### GCE Ingress and manual HTTPS
+To set up HTTPS with GCE Ingress, you must:
+1. offload the proxy, enable `https` (and ideally, change it from 
```

Comment:
No need to enable https, it's handled at Ingress rather than `proxy`.

By setting `enabled: true, type: offload`, the `proxy` service

- opens port 443, but _ingress_ [never send request to it](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/eebc19c3c4a5e7d3da68b026c31ebd99d0f048b1/jupyterhub/templates/ingress.yaml#L23)
- is expected to terminate SSL, but GKE _service_ has no such feature

so it's useless.

ref: https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/master/doc/source/security.md#off-loading-ssl-to-a-load-balancer

<details>
<summary>Click here to see how `proxy` service works for each https type</summary>

Legend: `[*]` is where SSL termination happens.

## enabled: false

```
  svc: proxy                                          pod: proxy
  ┌────────────┐                                      ┌────────────┐
─ 80 ──────────┼───────────────────────────────────── 8000         │
  │            │                                      │            │
  └────────────┘                                      └────────────┘
```

## enabled: true, type: manual/secret

```
  svc: proxy                                          pod: proxy
  ┌────────────┐                                      ┌────────────┐
─ 80 ──────────┼───────────────────────────────────── 8000         │
─ 443 ─────────┼───────────────────────────────────── 8443─[*]→ redirect
  └────────────┘                                      └────────────┘
```

## enabled: true, type: offload

```
  svc: proxy                                          pod: proxy
  ┌────────────┐                                      ┌────────────┐
─ 80 ───────┬──┼───────────────────────────────────── 8000         │
─ 443 ──[*]─┘  │                                      │            │
  └────────────┘                                      └────────────┘
```

## enabled: true, type: auto

```
  svc: proxy                                          pod: proxy
  ┌────────────┐                                      ┌────────────┐
─ 80 ──────────┼─┐                                  ┌ 8000         │
─ 443 ─────────┼┐│ pod: autohttps    svc: autohttps │ │            │
  └────────────┘││ ┌────────────┐    ┌────────────┐ │ └────────────┘
                │└ 80 ───────┬──┼─── 8000 ────────┼─┘
                └─ 443 ──[*]─┘  │    │            │
                   └────────────┘    └────────────┘
```

</details>

Reply:
Are you saying that I should remove **enabled: true, type: offload** in the **proxy** section and set up https following the directions below? 
ref: [https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/master/doc/source/security.md#off-loading-ssl-to-a-load-balancer](url)

Reply:
If you want to use GCP's HTTPS load balancer,

> remove **enabled: true, type: offload** in the **proxy** section

Yes. (actually no problem to set that, but it's pointless)

> set up https following the directions below?

No, that direction is not for you. It's for who want to set up https using service of `type: LoadBalancer`, such as AWS ELB.

Note that k8s has 2 ways of representations of load balancers:

1. Ingress
    - e.g.: GCP's HTTP(S) LB
2. Service of `type: LoadBalancer`
    - e.g.: AWS's ELB/NLB, GCP's TCP LB

Reply:
Is there any specific instructions or any documentation that I can follow in order to build load balancer for k8s? Thanks for the help!

Reply:
`build loadbalancer`? What do you mean by that? There is  a service of `type: LoadBalancer` or an `Ingress` resource, which can also be a loadbalancer (usually used for SSL purposes or to have single loadbalancer for many services).

Reply:
I mean that I want to let my JupyterHub site to have a loadbalancer. What should I do? I read the section **off-loading SSL to a load balancer** from this website [https://zero-to-jupyterhub.readthedocs.io/en/latest/security.html?highlight=load%20balance](url), but I am still confused on how to set up loadbalancer in the Google cloud platform. Any help would be appreciated.

Reply:
It's not necessary. E.g. if you are in a trusted network (e.g. at home) you don't need SSL (at least I wouldn't bother).


The reason for it for me was that we had SSL handled by Ingress, therefore there is no reason why to let SSL on for anything beyond Ingress. That's why it's off-loading. The `proxy` is by default a proper k8s `Service` resource with type `LoadBalancer`, if that's what you want. Then there is a k8s Ingress resource, which as different type of a LoadBalancer. It's a bit confusing.